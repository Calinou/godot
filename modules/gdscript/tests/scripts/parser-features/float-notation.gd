func test():
	# The following floating-point notations are all valid:
	assert(is_equal_approx(123., 123))
	assert(is_equal_approx(.123, 0.123))
	assert(is_equal_approx(.123e4, 1230))
	assert(is_equal_approx(123.e4, 1.23e6))
	assert(is_equal_approx(.123e-1, 0.0123))
	assert(is_equal_approx(123.e-1, 12.3))

	# Same as above, but with negative numbers.
	assert(is_equal_approx(-123., -123))
	assert(is_equal_approx(-.123, -0.123))
	assert(is_equal_approx(-.123e4, -1230))
	assert(is_equal_approx(-123.e4, -1.23e6))
	assert(is_equal_approx(-.123e-1, -0.0123))
	assert(is_equal_approx(-123.e-1, -12.3))

	# Same as above, but with explicit positive numbers (which is redundant).
	assert(is_equal_approx(+123., +123))
	assert(is_equal_approx(+.123, +0.123))
	assert(is_equal_approx(+.123e4, +1230))
	assert(is_equal_approx(+123.e4, +1.23e6))
	assert(is_equal_approx(+.123e-1, +0.0123))
	assert(is_equal_approx(+123.e-1, +12.3))
