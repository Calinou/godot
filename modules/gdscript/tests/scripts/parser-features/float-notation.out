GDTEST_OK
>> WARNING
>> Line: 3
>> ASSERT_ALWAYS_TRUE
>> Assert statement is redundant because the expression is always true.
>> WARNING
>> Line: 4
>> ASSERT_ALWAYS_TRUE
>> Assert statement is redundant because the expression is always true.
>> WARNING
>> Line: 5
>> ASSERT_ALWAYS_TRUE
>> Assert statement is redundant because the expression is always true.
>> WARNING
>> Line: 6
>> ASSERT_ALWAYS_TRUE
>> Assert statement is redundant because the expression is always true.
>> WARNING
>> Line: 7
>> ASSERT_ALWAYS_TRUE
>> Assert statement is redundant because the expression is always true.
>> WARNING
>> Line: 8
>> ASSERT_ALWAYS_TRUE
>> Assert statement is redundant because the expression is always true.
>> WARNING
>> Line: 11
>> ASSERT_ALWAYS_TRUE
>> Assert statement is redundant because the expression is always true.
>> WARNING
>> Line: 12
>> ASSERT_ALWAYS_TRUE
>> Assert statement is redundant because the expression is always true.
>> WARNING
>> Line: 13
>> ASSERT_ALWAYS_TRUE
>> Assert statement is redundant because the expression is always true.
>> WARNING
>> Line: 14
>> ASSERT_ALWAYS_TRUE
>> Assert statement is redundant because the expression is always true.
>> WARNING
>> Line: 15
>> ASSERT_ALWAYS_TRUE
>> Assert statement is redundant because the expression is always true.
>> WARNING
>> Line: 16
>> ASSERT_ALWAYS_TRUE
>> Assert statement is redundant because the expression is always true.
>> WARNING
>> Line: 19
>> ASSERT_ALWAYS_TRUE
>> Assert statement is redundant because the expression is always true.
>> WARNING
>> Line: 20
>> ASSERT_ALWAYS_TRUE
>> Assert statement is redundant because the expression is always true.
>> WARNING
>> Line: 21
>> ASSERT_ALWAYS_TRUE
>> Assert statement is redundant because the expression is always true.
>> WARNING
>> Line: 22
>> ASSERT_ALWAYS_TRUE
>> Assert statement is redundant because the expression is always true.
>> WARNING
>> Line: 23
>> ASSERT_ALWAYS_TRUE
>> Assert statement is redundant because the expression is always true.
>> WARNING
>> Line: 24
>> ASSERT_ALWAYS_TRUE
>> Assert statement is redundant because the expression is always true.
